test/ValidationFramework/Validation.test.cpp

```diff
@@ -0,0 +1,51 @@
+/// Apache License 2.0
+
+#include <iostream>
+
+#include <ValidationFramework/Validation.hpp>
+
+#include <Global.test.hpp>
+
+TEST_F(Validation, ValidationRunner)
+{
+    for (const auto& testCase_ : testCases_)
+    {
+        // Run test
+        // testing::internal::CaptureStdout();
+        std::cout << "Running test: " << testCase_.testName_ << std::endl;
+        // std::cout << "Comparisons with: " << testCase_.comparisons_ << std::endl;
+        // std::cout << "Tolerance: " << scenario_.tolerance_ << std::endl;
+        // EXPECT_FALSE(testing::internal::GetCapturedStdout().empty());
+
+        // // Call mission sequence app to set up the scenario
+        // const MissionSequence missionSequence = {scenario_.testName_};
+
+        // // Compare with reference
+        // for (const auto& index : len(compareWith_))
+        // {
+        //     // Run scenario
+        //     const std::vector<ComparisonResults> comparisonResults =
+        //     missionSequence.run(scenario_.compareWith_[index]);
+
+        //     // Assert on comparison results
+        //     EXPECT_GT(scenario_.tolerance_[index], comparisonResults[index]);
+        // }
+    }
+}
+
+// Alternative approach, validation test definitions in a separate yaml file and parsed in:
+// std::ifstream configFile("validation/ValidationConfig.yaml");
+// YAML::Node config = YAML::Load(configFile);
+
+// // Parse validation config file
+// for (const auto& scenario : config["scenarios"])
+// {
+//     TestCase testCase;
+//     testCase.testName = scenario["test"].as<std::string>();
+//     for (const auto& compareWith : scenario["compare-with"])
+//     {
+//         testCase.compareWith.push_back(compareWith.as<std::string>());
+//     }
+//     testCase.tolerance = scenario["tolerance"].as<double>();
+//     validationConfig_.scenarios.push_back(testCase);
+// }
```

Comment:
Let's remove this for now as we go for approach 1. Let's use the proposal or a separate MR later to show how we would go from the intermediate step we are taking right now to this next step